network/p2p/middleware/middleware.go

```diff
@@ -237,7 +237,7 @@ func (m *Middleware) NewPingService(pingProtocol protocol.ID, provider network.P
 func (m *Middleware) peerIDs(flowIDs flow.IdentifierList) peer.IDSlice {
 	result := make([]peer.ID, len(flowIDs))
```

Comment:
agree with the usage of `append` as the default pattern. However, than you need to instantiate the slice here also with length of zero and capacity len(flowIDs).
```suggestion
	result := make([]peer.ID, 0, len(flowIDs))
```

Reply:
https://github.com/onflow/flow-go/pull/3748/commits/1c1af46f6e71c00de7db09eceae43f927cc08ebd